frontend/src/index.tsx

```diff
@@ -111,7 +111,19 @@ const logoutLink = onError(({ networkError, graphQLErrors }: ErrorResponse) => {
       );
       return message;
     });
-    showError("Please check for errors and try again", messages.join(" "));
+    if (messages.includes("Error submitting test")) {
```

Comment:
I think you should be able to return the error message directly from the backend (in GraphQlConfig) rather than catching it directly here, right?

Reply:
I think the issue I was running into was both the header for the error message and content of the error message would be different than our generic "Please check for errors and try again" error message content. 

With that in mind, what would be the best thing to do since it doesn't look like the `GraphQLError` type supports two separate error messages. Should I send the entire error message from the backend and split that error message on the frontend? Let me know if I'm thinking about this way too complicatedly 😅

Reply:
Hmm ok I see the issue! I think you should indeed send the whole message (header + body) from the backend and just split it on the frontend. I would actually love to see a generic-ish implementation so we could reuse the same frontend logic, if that's feasible :) I'm thinking something like
`new GraphQlError("header: $header, body: $body");` on the backend, and then splitting/substring-ing the error message on the frontend into a separate header and body. Let me know if that approach makes sense!

Reply:
Would these frontend changes cause any concern for backwards compatibility since we're updating the backend at the same time? I'm thinking it should be ok, in worst case the error printed out by the frontend might just look a little funny if the frontend code hasn't deployed with the backend.

Reply:
That's a good point! But I don't think backwards compat is too much of an issue here - like you said, worst case scenario is that the frontend error message is a little wonky. 

The backwards compat issue is semi-resolved in prod, I think - IIRC, it checks every ~15 minutes to make sure the commit hash is the same as the most recent version.

Reply:
Thanks @emmastephenson! That makes a lot of sense! Thanks for that reminder, George. Glad this change won't be too disruptive so I'll wrap all these changes in one PR! 😄